endurance-page-cache.php

```diff
@@ -585,19 +595,50 @@ public function purge_cdn_single( $pattern = '' ) {
 		/**
 		 * Ensure that a URI isn't purged more than once per minute.
 		 *
-		 * @param string $value URI being purged
+		 * @param string $uri URI being purged
 		 *
-		 * @return bool
+		 * @return bool True if additional purges should be avoided, false otherwise.
 		 */
-		public function purge_throttle( $value ) {
-			$purged = get_transient( 'epc_purged_' . md5( $value ) );
-			if ( ( true === $purged || in_array( md5( $value ), $this->purged, true ) ) && false === $this->force_purge ) {
-				return true;
+		public function should_throttle( $uri ) {
+
+			$should_throttle = false;
+
+			$this->should_update_purges = true;
+
+			$hash = md5( $uri );
+
+			if ( array_key_exists( $hash, $this->purged ) ) {
+				if ( $this->purged[ $hash ] < time() ) {
+					unset( $this->purged[ $hash ] );
+				} else {
+					$should_throttle = true;
+				}
 			}
-			set_transient( 'epc_purged_' . md5( $value ), time(), 60 );
-			$this->purged[] = md5( $value );
 
-			return false;
+			$this->purged[ $hash ] = time() + 60;
```

Comment:
This is always going to push the hash time out by 60 seconds, whether it's been throttled or not. So as long as these purge requests keep firing within the minute it keeps pushing them out. 

Should we have a hard 60 second cap, and allow purges after that, or keep pushing it out like this every time if a purge fires within that limit?